### PR TITLE
[SPARK-44466][SQL] Exclude configs starting with `SPARK_DRIVER_PREFIX` and `SPARK_EXECUTOR_PREFIX` from modifiedConfigs

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?

This PR excludes configs starting with `SPARK_DRIVER_PREFIX` and `SPARK_EXECUTOR_PREFIX` from modifiedConfigs.

### Why are the changes needed?

To make `SQL / DataFrame Properties` excluding these properties because some configs [might supplemented](https://github.com/apache/spark/blob/caa3df48d94ff2e7c824a87acf51ab4978e18098/core/src/main/scala/org/apache/spark/SparkContext.scala#L422-L423):
image

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Unit test.

### CODE REVIEW

**Comment by wangyum** [comment]
cc @ulysses-you @gengliangwang

File: sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala [hunk 1]
```diff
@@ -1103,7 +1103,9 @@ object SparkSession extends Logging {
         loadExtensions(extensions)
         applyExtensions(sparkContext, extensions)
 
-        session = new SparkSession(sparkContext, None, None, extensions, options.toMap)
+        // Some config might supplemented. For example: spark.driver.extraJavaOptions.
+        val supplementedOptions = options.map { case (k, _) => k -> sparkContext.conf.get(k) }
```
Comment (ulysses-you) [inline sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala:1107]: Does `sparkContext.conf` contain all entries in options ? What if there is an active SparkContext before creating SparkSession.

Comment (wangyum) [reply sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala]: Makes it exclude configs starting with `SPARK_DRIVER_PREFIX` and `SPARK_EXECUTOR_PREFIX`.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala [hunk 2]
```diff
@@ -97,7 +98,11 @@ object SQLExecution {
 
       val globalConfigs = sparkSession.sharedState.conf.getAll.toMap
       val modifiedConfigs = sparkSession.sessionState.conf.getAllConfs
-        .filterNot(kv => globalConfigs.get(kv._1).contains(kv._2))
+        .filterNot { case (key, value) =>
+          key.startsWith(SPARK_DRIVER_PREFIX) ||
+            key.startsWith(SPARK_EXECUTOR_PREFIX) ||
+            globalConfigs.get(key).contains(value)
```
Comment (ulysses-you) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala:104]: make sense to me as these configs are kind of static configs, but cc @HyukjinKwon I'm not sure if python side depends on any configs with these prefix at runtime

**Comment by wangyum** [comment]
cc @cloud-fan @HyukjinKwon

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala (hunk 2 above)
Comment (HyukjinKwon) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala:104]: I think the fix is fine.

**Review by HyukjinKwon — APPROVED** [review]
I am okay although this is kind of a bandaid fix. Ideally we should just filter `spark.sql` and show them but .. that might be breaking changes e.g., external datasource using the configs.

**Review by ulysses-you — APPROVED** [review]

**Comment by wangyum** [comment]
Merged to master and branch-3.5.

**Pull request closed without merging**